### PR TITLE
Serialize and load JSON Light .shadow property

### PR DESCRIPTION
**Pull request by satori99**
- Add .toJSON() method to LightShadow
- Light .toJSON() writes .shadow property
- Parse .shadow property in DirectionLight and Spotlight JSON data

### CODE REVIEW

**Comment by satori99** [comment]
A DirectionalLight that casts a shadow would be serialized something like this;

            {
                "uuid": "24b1e136-e0eb-5fe2-b08b-c2a705bad1c2",
                "type": "DirectionalLight",
                "color": 16777164,
                "intensity": 1.5,
                "castShadow": true,
                "shadow": {
                    "mapSize": [512,1024],
                    "camera": {
                        "type": "OrthographicCamera",
                        "top": 7,
                        "bottom": -1.5,
                        "left": -2.5,
                        "right": 2.5,
                        "near": 8,
                        "far": 32
                    }
                },
                "matrix": [1,0,0,0,0,1,0,0,0,0,1,0,8,16,10,1]
            }

File: src/lights/Light.js [hunk 1]
```diff
@@ -50,6 +50,8 @@ Light.prototype = Object.assign( Object.create( Object3D.prototype ), {
 		if ( this.decay !== undefined ) data.object.decay = this.decay;
 		if ( this.penumbra !== undefined ) data.object.penumbra = this.penumbra;
 
+		if ( this.castShadow && this.shadow !== undefined ) data.object.shadow = this.shadow.toJSON();
```
Comment (mrdoob) [inline src/lights/Light.js]: I wonder if the `this.castShadow` check is needed. Thinking about the editor case... One could set the shadow parameters but have the `castShadow` temporarily `false`. The shadow parameters would then get lost.

**Comment by satori99** [comment]
The .shadow (LightShadow) property of DirectionalLights and SpotLights is always created by the lights constructor. Even when shadows are not used at all. So this would always be exported I think.

If its acceptable to always export default shadow settings, then I can remove that condition.

**Comment by satori99** [comment]
I removed the castShadow check. I thought about it a bit more, and I think you are right. It doesn't really matter if shadows are used in a scene or not, as that is a renderer setting anyway.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**